pytest/alignment_test.py

```diff
@@ -0,0 +1,182 @@
+import pytest
+
+import arkouda as ak
+
+
+def test_search_interval():
+    expected_result = [2, 5, 4, 0, 3, 1, 4, -1, -1]
+    lb = [0, 10, 20, 30, 40, 50]
+    ub = [9, 19, 29, 39, 49, 59]
+    v = [22, 51, 44, 1, 38, 19, 40, 60, 100]
+
+    # test int64
+    lower_bound = ak.array(lb)
+    upper_bound = ak.array(ub)
+    vals = ak.array(v)
+    interval_idxs = ak.search_intervals(vals, (lower_bound, upper_bound))
+    assert expected_result == interval_idxs.to_list()
+
+    # test uint64
+    lower_bound = ak.array(lb, dtype=ak.uint64)
+    upper_bound = ak.array(ub, dtype=ak.uint64)
+    vals = ak.array(v, dtype=ak.uint64)
+    interval_idxs = ak.search_intervals(vals, (lower_bound, upper_bound))
+    assert expected_result == interval_idxs.to_list()
+
+    # test uint64
+    lower_bound = ak.array(lb, dtype=ak.float64)
+    upper_bound = ak.array(ub, dtype=ak.float64)
+    vals = ak.array(v, dtype=ak.float64)
+    interval_idxs = ak.search_intervals(vals, (lower_bound, upper_bound))
+    assert expected_result == interval_idxs.to_list()
+
+
+def test_multi_array_search_interval():
+    # Added for Issue #1548
+    starts = (ak.array([0, 10, 20]), ak.array([0, 10, 20]))
+    ends = (ak.array([4, 14, 24]), ak.array([4, 14, 24]))
+    vals = (ak.array([3, 13, 23]), ak.array([23, 13, 3]))
+    ans = [-1, 1, -1]
+    assert ans == ak.search_intervals(vals, (starts, ends), hierarchical=False).to_list()
+    assert ans == ak.interval_lookup((starts, ends), ak.arange(3), vals).to_list()
+
+    vals = (ak.array([23, 13, 3]), ak.array([23, 13, 3]))
+    ans = [2, 1, 0]
+    assert ans == ak.search_intervals(vals, (starts, ends), hierarchical=False).to_list()
+    assert ans == ak.interval_lookup((starts, ends), ak.arange(3), vals).to_list()
+
+    vals = (ak.array([23, 13, 33]), ak.array([23, 13, 3]))
+    ans = [2, 1, -1]
+    assert ans == ak.search_intervals(vals, (starts, ends), hierarchical=False).to_list()
+    assert ans == ak.interval_lookup((starts, ends), ak.arange(3), vals).to_list()
+
+    # test hierarchical flag
+    starts = (ak.array([0, 5]), ak.array([0, 11]))
+    ends = (ak.array([5, 9]), ak.array([10, 20]))
+    vals = (ak.array([0, 0, 2, 5, 5, 6, 6, 9]), ak.array([0, 20, 1, 5, 15, 0, 12, 30]))
+
+    assert ak.search_intervals(vals, (starts, ends), hierarchical=False).to_list() == [
+        0,
+        -1,
+        0,
+        0,
+        1,
+        -1,
+        1,
+        -1,
+    ]
+    search_intervals_hierarchical = ak.search_intervals(vals, (starts, ends)).to_list()
+    assert search_intervals_hierarchical == [0, 0, 0, 0, 1, 1, 1, -1]

```

Comment:
if we could do something like `search_intervals_hierarchical` for the `non-hierarchical` case to avoid the black getting so wild with the formatting that would sweet

```python
search_int_non_hierarch = ak.search_intervals(vals, (starts, ends), hierarchical=False).to_list()
assert search_int_non_hierarch == [0, -1, 0, 0, 1, -1, 1, -1]
search_intervals_hierarchical = ak.search_intervals(vals, (starts, ends)).to_list()
assert search_intervals_hierarchical == [0, 0, 0, 0, 1, 1, 1, -1]
```

Reply:
Added your suggested edit